Lesson16/src/main.js

```diff
@@ -159,25 +164,110 @@ function addImage({ image }) {
 }
 
 async function createClickedTabContent(target) {
+  const tabContent = document.getElementById("js-tab-content");
   const targetId = await target.id;
-  const json = await tryGetData(API[targetId]);
+  const json = await tryGetData(tabContent, API[targetId]);
   createArticleElements(json);
   addImage(json);
 }
 
-async function tryGetData(resource) {
-  addLoading();
+async function tryGetData(toAppend, resource) {
+  addLoading(toAppend);
   try {
     return await getJsonOrError(resource);
   } catch (e) {
-    addErrorMessage(e);
+    addErrorMessage(e, toAppend);
   } finally {
     removeLoading();
   }
 }
 
+function addModal() {
+  const modal = createElementWithClass("div", "modal");
+  const modalInner = createElementWithClass("div", "modal__inner")
+  modal.id = "js-modal";
+  modalInner.id = "js-modal-inner";
+  modal.appendChild(modalInner);
+  document.querySelector("body").appendChild(modal);
+  addModalCloseIcon(modal);
+}
+
+function addModalCloseIcon(toAppend) {
+  const modalCloseWrapper = createElementWithClass("div", "modal__close-wrapper");
+  const icon = document.createElement("img");
+  icon.src = "./img/cross-icon.svg";
+  modalCloseWrapper.appendChild(icon);
+  toAppend.appendChild(modalCloseWrapper);
+  setClickEventForModalClose(icon);
+}
+
+function addOverLay() {
+  const overLay = createElementWithClass("div", "over-lay");
+  overLay.id = "js-over-lay";
+  document.querySelector("body").appendChild(overLay);
+  setClickEventForModalClose(overLay);
+}
+
+function setClickEventForModalClose(target) {
+  target.addEventListener("click", () => {
+    document.getElementById("js-modal-inner").textContent = "";
+    document.getElementById("js-modal").classList.remove("is-modal-open");
+    document.getElementById("js-over-lay").classList.remove("is-overlay-display");
+  });
+}
+
+function openModalAndOverLay() {
+  document.getElementById("js-modal").classList.add("is-modal-open");
+  document.getElementById("js-over-lay").classList.add("is-overlay-display");
+}
+
+function setClickEventInCommentIcon(target) {
+  target.addEventListener("click", (e) => {
+    const toAppendElement = document.getElementById("js-modal-inner");
+    const targetParentId = e.currentTarget.closest("li").id;
+    tryCommentContentCreate(targetParentId, toAppendElement);
+    openModalAndOverLay();
+  });
+}
+
+async function getClickedArticleData(resource, toAppend) {
+  return await tryGetData(toAppend, articleAPI[resource]);
+}
+
+async function tryCommentContentCreate(targetId, toAppend) {
+  const responseData = await getClickedArticleData(targetId, toAppend);
+  const { comment } = responseData;
+  createAndAddCommentContent(comment, toAppend);
+}
```

Comment:
```suggestion
async function getComment(targetId, toAppend) {
  const responseData = await getClickedArticleData(targetId, toAppend);
  const { comment } = responseData;
  return comment
}
```

Reply:
I'm including it in the commit with the fix above.
Please confirm.
6dcb4d0